File: 2021/day02/day02-01.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int h_pos=0,depth=0;
char command[3]={"forward","up","down"};

int main() {
    char line[11], str_command[10], str_value[4];
    int int_value;
    FILE* file = fopen("/Users/ricardovieira/github/adventofcode/2021/day02/input.txt", "r");

    while (fgets(line, 10, file)) {
        strcpy(str_command, strsep(line, " "));
        strcpy(str_value, strsep(NULL, " "));
        int_value = atoi(str_value);
        if (str_command == command[0]) h_pos += atoi(int_value);
        if (str_command == command[1]) depth += atoi(int_value);
        if (str_command == command[2]) depth -= atoi(int_value);
    }
    printf("%i\n", h_pos*depth);
    fclose(file);
}

File: 2022/day02/day02-01.py
totalscore=score=i=0
for line in open('/Users/ricardovieira/github/adventofcode/2022/day02/input.txt'):
    score=0
    i+=1
    if line.split()[0] == "A": # Rock
        score+=1
        if line.split()[1] == "X": # Rock
            score+=3
        elif line.split()[1] == "Y": # Paper
            score+=0
        else: # Scissor
            score+=6
    elif line.split()[0] == "B": # Paper
        score+=2
        if line.split()[1] == "X": #Rock
            score+=6
        elif line.split()[1] == "Y": # Paper
            score+=3
        else: # Scissor
            score+=0
    else: # Scissor
        score+=3
        if line.split()[1] == "X": #Rock
            score+=0
        elif line.split()[1] == "Y": # Paper
            score+=6
        else: # Scissor
            score+=3
 #   print(score)
    totalscore+=score
    print(str(i)+': '+line+str(score)+','+str(totalscore))
#    print(totalscore)
print()
print(totalscore)
